### PR TITLE
Update documentation for 0.3 release

### PR DESCRIPTION
**Pull request by sameerz**
Update documentation for 0.3 

Partially addresses #1257

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+title: Download
+nav_order: 10
+---
+
+## Release v0.3.0
+This release includes additional performance improvements, for example, use of per thread default
+stream to make more efficient use of the GPU.  There is also further support for adaptive query
+execution, with more rewritten query plans now able to run on the GPU.  There are performance
+improvements for reading small Parquet files.  RAPIDS Shuffle with UCX has been updated to UCX
+1.9.0.
```
Comment (jlowe) [inline docs/download.md:12]: Suggestion: bullet points seem apropos here, e.g.:
```suggestion
This release includes additional performance improvements, including:
- using per thread default stream to make more efficient use of the GPU
- supporting Spark's adaptive query execution, with more rewritten query plans now able to run on the GPU
- accelerating reads of many small Parquet files
- updating RAPIDS Shuffle with UCX to UCX 1.9.0
```

File: docs/download.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+Additional functionality includes Parquet reading for lists and structs, lead/lag for windows, and
```
Comment (jlowe) [inline docs/download.md:14]: Similarly may want a "new functionality" bullet list here.

File: docs/download.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+greatest/least operators among others.
+
+The release is supported on Apache Spark 3.0.0, 3.0.1, Databricks 7.3 ML LTS and Google Cloud
+Platform Dataproc 2.0.
+
+GPU reading of cached dataframes is enabled but only effective for Spark 3.1, for those who
```
Comment (jlowe) [inline docs/download.md:20]: This requires the user configure the batched cache serializer, correct?  If we're going to mention it here we may want to point them to the documentation on how to do this.  AFAIK it won't "just work" if they use Spark 3.1 with the plugin.

File: docs/download.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+For a detailed list of changes, please refer to the
+[CHANGELOG](https://github.com/NVIDIA/spark-rapids/blob/main/CHANGELOG.md). 
+
+Hardware Requirements: 
+
+	GPU Architecture: NVIDIA Pascal™ or better (Tested on V100, T4 and A100 GPU)
+	
+Software Requirements:
+
+	OS: Ubuntu 16.04 & gcc 5.4 OR Ubuntu 18.04/CentOS 7 & gcc 7.3
```
Comment (jlowe) [inline docs/download.md:34]: This project has no native code and therefore doesn't need a GCC version.
```suggestion
	OS: Ubuntu 16.04, Ubuntu 18.04, or CentOS 7
```

File: docs/download.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
```
Comment (jlowe) [inline docs/download.md:80]:
```suggestion
	OS: Ubuntu 16.04, Ubuntu 18.04, or CentOS 7
```

File: docs/download.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+## Release v0.1.0
+
+Hardware Requirements: 
+   
+    GPU Architecture: NVIDIA Pascal™ or better (Tested on V100 and T4 GPU)
+
+Software Requirements: 
+
+    OS: Ubuntu 16.04 & gcc 5.4 OR Ubuntu 18.04/CentOS 7 & gcc 7.3
```
Comment (jlowe) [inline docs/download.md:104]:
```suggestion
    OS: Ubuntu 16.04 , Ubuntu 18.04, or CentOS 7
```

File: docs/download.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
+---
+layout: page
+title: Download
+nav_order: 10
```
Comment (jlowe) [inline docs/download.md:4]: I suggest making this a much lower nav order, meaning it's much higher on the list and therefore easier to see.  Placing it last makes it look like we're trying to hide it and it's an afterthought.  Just after or before the "Getting Started" section seems like a logical order to me.

**Review by jlowe — COMMENTED** [review]
Looks pretty good, mostly suggestions on making the download links more prominent and using bullet lists when iterating things in the release.

**Comment by sameerz** [comment]
The supported_ops.md file was hand generated in branch-0.3 per #1268.  I will follow up with an update to https://github.com/NVIDIA/spark-rapids/blob/branch-0.4/sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala in branch-0.4.

**Review by jlowe — DISMISSED** [review]

**Comment by sameerz** [comment]
Build

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Pull request merged**